### ISSUE TITLE
INDArray.neq() is no-op on CUDA backend

### ISSUE THREAD

**Issue by EdeMeijer** [deeplearning4j/nd4j#1474]
The result of 
```java
Nd4j.rand(2, 2).neq(1)
```
Is just a random 2x2 matrix. On native backend, it gives the correct boolean matrix.

**Comment by raver119** [comment]
Can i see stack trace please?

**Comment by EdeMeijer** [comment]
There's no stack trace, there's no error. Its just that
```java
INDArray rnd = Nd4j.rand(2, 2);
System.out.println(rnd.equals(rnd.neq(1)));
```
outputs `true`

`neq()` doesn't change anything to the matrix it's applied to

**Comment by EdeMeijer** [comment]
Same seems go to go for `lt()` and `gt()`

**Comment by EdeMeijer** [comment]
Interestingly,
```java
System.out.println(Nd4j.rand(2, 2).gti(0.5));
```

outputs a binary matrix while

```java
System.out.println(Nd4j.rand(2, 2).gt(0.5));
```
doesn't.

Seems like the original matrix is returned instead of the one copied and modified. But again, only on CUDA.

**Comment by EdeMeijer** [comment]
Hmm, I'm not really sure why, but whatever `dup()` produces seems to be immune to some operations, like `neq`, `lt` and `gt`.

**Comment by raver119** [comment]
Fast fix was applied, proper fix will be implemented after 0.7.1 release, together with grid fix of ScalarMax/ScalarMin. The same cause is behind of these issues.

**Comment by raver119** [comment]
p.s. Tests to be added for all those methods.
